src/corecel/sys/Stopwatch.hh

```diff
@@ -61,9 +68,24 @@ double Stopwatch::operator()() const
 {
     using DurationSec = std::chrono::duration<double>;
 
-    auto duration = Clock::now() - start_;
+    auto duration = Stopwatch::now() - start_;
     return std::chrono::duration_cast<DurationSec>(duration).count();
 }
 
+inline auto Stopwatch::now() -> TimePoint
+{
+    static bool const result = [] {
```

Comment:
best to avoid `static` storage duration in `inline` functions because it can make trouble for the linker... where else is the stopwatch being called that is adding this level of overhead? I guarded the timer in `celer-sim` 🤔 if it's just the "action" timing that counts, maybe we should make the counter there locally dependent on the `sync` variable (which, if off, disables GPU action timing)?

Reply:
Have you noticed static storage in inline function being an issue before? By the standard the object is shared between all definitions.

The call in ActionSequence::execute is the problem, on CPU it will always call the Stopwatch. Enabling it only when we have the sync option on could be an alternative.

Reply:
I've been burned before by *template* static inlines before... but compilers/linkers were more primitive and we could've been doing something subtly wrong.

I think it would be a better solution to suppress the timing at a higher level than the stopwatch though, especially since with "sync" off for GPU, the `["time"]["action"]` block is empty, whereas for CPU it would show all zeros. Perhaps we could rename `sync` to `action_times`?

Reply:
ah yeah, with template inline static is more tricky since each instantiation would define different functions.

I renamed sync to action_times, reverted the changes in Stopwatch, and only measure time when requested on host.